src/main/java/org/openstreetmap/atlas/utilities/archive/Extractor.java

```diff
@@ -182,44 +182,48 @@ public Extractor extract(final File inputFile) throws ArchiveException, IOExcept
         }
 
         fireArchiveStarted();
-        final ZipFile file = new ZipFile(inputFile);
-        for (final ZipArchiveEntry current : Collections.list(file.getEntries()))
+        try (final ZipFile file = new ZipFile(inputFile))
         {
-            final File outputFile = new File(this.outputDirectory, current.getName());
-
-            if (shouldSkip(outputFile))
-            {
-                fireItemSkipped(outputFile);
-            }
-            else if (outputFile.exists() && this.skipExisting)
+            for (final ZipArchiveEntry current : Collections.list(file.getEntries()))
             {
-                fireItemSkipped(outputFile);
+                final File outputFile = new File(this.outputDirectory, current.getName());
+
+                if (shouldSkip(outputFile))
+                {
+                    fireItemSkipped(outputFile);
+                }
+                else if (outputFile.exists() && this.skipExisting)
+                {
+                    fireItemSkipped(outputFile);
+                }
+                else if (current.isDirectory())
+                {
+                    continue;
+                }
+                else
+                {
+                    try (final BufferedOutputStream bos = new BufferedOutputStream(
+                            new FileOutputStream(outputFile));
+                            final InputStream inputStream = file.getInputStream(current))
+                    {
+                        outputFile.getParentFile().mkdirs();
+                        NotifyingIOUtils.copy(inputStream, bos,
+                                new Progress(outputFile, current.getSize()));
+                        IOUtils.closeQuietly(inputStream);
```

Comment:
Are the two `closeQuietly` calls still needed? `close()` is implicitly called by the try-with-resource block.

Reply:
Hmm good point. Maybe worth testing with and without the calls.

Reply:
Ok, let me add a unit test